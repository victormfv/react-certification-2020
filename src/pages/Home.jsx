import React from 'react';
import VideoList from '../components/VideoList';
import VideoPlayer from '../components/VideoPlayer';
import { useVideos } from '../state/VideosProvider';
// <div className="flex flex-wrap flex-row justify-center p-20 min-h-screen ">

function Home() {
  
  const { playing} =useVideos()

  return (
    <div className="flex flex-row justify-center min-w-full ">
        <div className="w-4/5 flex flex-row justify-center border p-10" >
           <div className="flex border flex-col h-youtube-l w-youtube-l">
             <VideoPlayer/>
             </div>
        </div>
        <div className="flex flex-col border p-10">
          <VideoList />
        </div>
    </div>
  );

}

export default Home;
