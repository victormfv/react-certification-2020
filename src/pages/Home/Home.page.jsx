import React, { useRef,useState, useEffect } from 'react';
import { Link} from 'react-router-dom';
import { useAuth } from '../../providers/Auth';
import './Home.styles.css';
import VideoCard from '../../components/VideoCard';
import styled from 'styled-components';
const dataJson ="https://gist.githubusercontent.com/jparciga/1d4dd34fb06ba74237f8966e2e777ff5/raw/f3af25f1505deb67e2cc9ee625a633f24d8983ff/youtube-videos-mock.json";

const Videos = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin: 0 auto;
  padding: 20px;
`;

function HomePage() {
    const [videos, setVideos] = useState(null);
    const sectionRef = useRef(null);
    const { authenticated } = useAuth();

    useEffect(() => {
    fetch(dataJson)
    .then(response => response.json())
    .then((jsonData) => {
      setVideos(jsonData.items);
    })
    .catch((error) => {
      console.error(error)
    });
  }, []);

  return (
    <section className="homepage" ref={sectionRef}>
      <h2>Welcome to the Challenge!</h2>
      {authenticated ? (
        <Videos>
         {videos? (videos.map((video)=>(
         <VideoCard key={video.etag} video={video.snippet} />))):""}
        </Videos>
      ) : (
        <Link to="/login">let me in â†’</Link>
      )}
    </section>
  );
}

export default HomePage;
