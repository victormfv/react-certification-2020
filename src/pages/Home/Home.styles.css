.homepage {
  text-align: center;
}

.homepage h2 {
  font-size: 3rem;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight: 300;
  line-height: 1;
  letter-spacing: -0.00833em;
}
