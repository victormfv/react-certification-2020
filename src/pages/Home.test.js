import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Home from './Home';

describe('Home Page', () => {

    it('renders correctly', () => {
      const tree = renderer.create(<Home/>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('Child component render is correct', () => {
      render(<Home/>);
      const i = screen.getByText(/Loading videos.../i);
      expect(i).toBeInTheDocument();
    });
  });