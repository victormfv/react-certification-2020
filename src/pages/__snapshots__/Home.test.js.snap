// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders correctly 1`] = `
<p>
  Could not fetch any data.
</p>
`;
